quora-service/src/main/java/com/upgrad/quora/service/business/QuestionService.java

```diff
@@ -19,4 +19,27 @@ public QuestionEntity getQuestionByUuid(String questionUuid) throws InvalidQuest
         }
         return questionEntity;
     }
+
+     /**
+   * Creates question in the DB if the accessToken is valid.
+   *
+   * @param accessToken accessToken of the user for valid authentication.
+   * @throws AuthorizationFailedException ATHR-001 - if user token is not present in DB. ATHR-002 if
+   *     the user has already signed out.
+   */
+  @Transactional(propagation = Propagation.REQUIRED)
+  public QuestionEntity createQuestion(QuestionEntity questionEntity, final String accessToken)
+      throws AuthorizationFailedException {
+    UserAuthEntity userAuthEntity = userAuthDao.getUserAuthByToken(accessToken);
+    if (userAuthEntity == null) {
+      throw new AuthorizationFailedException("ATHR-001", "User has not signed in");
+    } else if (userAuthEntity.getLogoutAt() != null) {
+      throw new AuthorizationFailedException(
+          "ATHR-002", "User is signed out.Sign in first to post a question");
+    }
```

Comment:
replace this code in AuthenticationService... have a look in validateToken method of AuthenticationService class